hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioChannel.java

```diff
@@ -70,18 +83,41 @@ public long lastWriteTimeMillis() {
     }
 
     @Override
-    public void flush() {
-        outboundPipeline.wakeup();
+    public void start() {
+        try {
+            // before starting the channel, the socketChannel need to be put in
+            // non blocking mode since that is mandatory for the NioChannel.
+            socketChannel.configureBlocking(false);
+            channelInitializer.initChannel(this);
+        } catch (Exception e) {
+            throw new HazelcastException("Failed to start " + this, e);
+        }
+        inboundPipeline.start();
+        outboundPipeline.start();
     }
 
     @Override
     protected void onClose() {
-        closeResource(inboundPipeline);
         closeResource(outboundPipeline);
     }
 
     @Override
     public String toString() {
-        return "NioChannel{" + localSocketAddress() + "->" + remoteSocketAddress() + '}';
+        try {
+            InetSocketAddress local = (InetSocketAddress) localSocketAddress();
+            InetSocketAddress remote = (InetSocketAddress) remoteSocketAddress();
+            String s = isClientMode() ? local.getPort() + "=>" + remote.getPort() : local.getPort() + "->" + remote.getPort();
+
+            // this is added for debugging so that 'client' and 'server' have a different indentation and are easy to recognize.
+            if (!isClientMode()) {
+                s = "                                                                                " + s;
+            }
+
+            Date date = new Date();
+            return date.getHours() + ":" + date.getMinutes() + ":" + date.getSeconds() + " " + s;
+        } catch (NullPointerException e) {
+            return "Better protection needed";
```

Comment:
?

Reply:
Method is for debugging only. Commented it out , so no longer an issue.